sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Util.scala

```diff
@@ -598,10 +598,28 @@ private[sql] object CatalogV2Util {
       //       data unchanged and let the data reader to return "exist default" for missing
       //       columns.
       val existingDefault = Literal(default.getValue.value(), default.getValue.dataType()).sql
-      f.withExistenceDefaultValue(existingDefault).withCurrentDefaultValue(default.getSql)
+      f.withExistenceDefaultValue(existingDefault).withCurrentDefaultValue(toSql(defaultValue))
     }.getOrElse(f)
   }
 
+  private def toSql(defaultValue: DefaultValue): String = {
+    if (defaultValue.getExpression != null) {
+      V2ExpressionUtils.toCatalyst(defaultValue.getExpression) match {
```

Comment:
Hmm, when it is possible that a default expression cannot be converted back to catalyst one?

Reply:
Dynamic expressions other than literals?

Reply:
Can we also add a test case that falls back to `defaultValue.getSql` when `toCatalyst` returns `None`?

Reply:
V2 expressions are quite flexible, users can represent a function with an arbitrary name. In practice, I expect it to be rare but you never know.

I'll add a test.

Reply:
Added more tests.